pkg/storage/sqlstorage/store.go

```diff
@@ -61,79 +42,7 @@ func (s *Store) Name() string {
 func (s *Store) Initialize(ctx context.Context) (bool, error) {
 	sharedlogging.GetLogger(ctx).Debug("Initialize store")
 
-	migrationsDir := fmt.Sprintf("migrations/%s", strings.ToLower(s.schema.Flavor().String()))
-	entries, err := fs.ReadDir(MigrationsFs, migrationsDir)
-	if err != nil {
-		return false, err
-	}
-
-	tx, err := s.schema.BeginTx(ctx, &sql.TxOptions{})
-	if err != nil {
-		return false, s.error(err)
-	}
-	defer func(tx *sql.Tx) {
-		_ = tx.Rollback()
-	}(tx)
-
-	modified := false
-	for _, m := range entries {
-		version := strings.TrimSuffix(m.Name(), ".sql")
-
-		sb := sqlbuilder.NewSelectBuilder()
-		sb.Select("version")
-		sb.From(s.schema.Table("migrations"))
-		sb.Where(sb.E("version", version))
-
-		// Does not use sql transaction because if the table does not exist, postgres will mark transaction as invalid
-		sqlq, args := sb.BuildWithFlavor(s.schema.Flavor())
-		row := s.schema.QueryRowContext(ctx, sqlq, args...)
-		var v string
-		if err = row.Scan(&v); err != nil {
-			sharedlogging.GetLogger(ctx).Debugf("%s", err)
-		}
-		if v != "" {
-			sharedlogging.GetLogger(ctx).Debugf("version %s already up to date", m.Name())
-			continue
-		}
-		modified = true
-
-		sharedlogging.GetLogger(ctx).Debugf("running migrations %s", m.Name())
-
-		b, err := migrations.ReadFile(path.Join(migrationsDir, m.Name()))
-		if err != nil {
-			sharedlogging.GetLogger(ctx).Errorf("%s", err)
-			return false, err
-		}
-
-		plain := strings.ReplaceAll(string(b), "VAR_LEDGER_NAME", s.schema.Name())
-		r := regexp.MustCompile(`[\n\t\s]+`)
-		plain = r.ReplaceAllString(plain, " ")
-
-		for i, statement := range strings.Split(plain, "--statement ") {
-			statement = strings.TrimSpace(statement)
-			if statement != "" {
-				sharedlogging.GetLogger(ctx).Debugf("running statement: %s", statement)
-				if _, err = tx.ExecContext(ctx, statement); err != nil {
-					err = errors.Wrapf(s.error(err), "failed to run statement %d", i)
-					err = errors.Wrapf(s.error(err), "statement: %s", statement)
-					sharedlogging.GetLogger(ctx).Errorf("%s", err)
-					return false, s.error(err)
-				}
-			}
-		}
-
-		ib := sqlbuilder.NewInsertBuilder()
-		ib.InsertInto(s.schema.Table("migrations"))
-		ib.Cols("version", "date")
-		ib.Values(version, time.Now().UTC().Format(time.RFC3339))
-		sqlq, args = ib.BuildWithFlavor(s.schema.Flavor())
-		if _, err = tx.ExecContext(ctx, sqlq, args...); err != nil {
-			sharedlogging.GetLogger(ctx).Errorf("failed to insert migration version %s: %s", version, err)
-			return false, s.error(err)
-		}
-	}
-
-	return modified, s.error(tx.Commit())
+	return CollectAndMigrate(ctx, s.schema)
```

Comment:
I'm not a big fan of such functions that does `ThisAndThat`. I think it is simpler to understand to split in two calls.
What do you think?